openmp/libomptarget/plugins-nextgen/common/src/PluginInterface.cpp

```diff
@@ -562,11 +565,12 @@ Error GenericKernelTy::launch(GenericDeviceTy &GenericDevice, void **ArgPtrs,
 
   // Record the kernel description after we modified the argument count and num
   // blocks/threads.
-  if (RecordReplay.isRecording()) {
-    RecordReplay.saveImage(getName(), getImage());
-    RecordReplay.saveKernelInput(getName(), getImage());
-    RecordReplay.saveKernelDescr(getName(), Ptrs.data(), KernelArgs.NumArgs,
-                                 NumBlocks, NumThreads, KernelArgs.Tripcount);
+  if (getRecordReplay().isRecording()) {
+    getRecordReplay().saveImage(getName(), getImage());
+    getRecordReplay().saveKernelInput(getName(), getImage());
+    getRecordReplay().saveKernelDescr(getName(), Ptrs.data(),
```

Comment:
Each call of `getRecordReplay` involves a lock. It should be called just once.

Reply:
Yeah, I should probably make a patch before this that just initializes it with the plugin initialization.